src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs

```diff
@@ -165,8 +165,20 @@ private async Task<FormulaValue> TryHandleSet(CallNode node, EvalVisitorContext
             var arg0 = node.Args[0];
             var arg1 = node.Args[1];
 
-            var newValue = await arg1.Accept(this, context).ConfigureAwait(false);
-
+            var newValue = await arg1.Accept(this, context).ConfigureAwait(false);
+
+            if (arg0.IRContext.IsMutation && arg0 is RecordFieldAccessNode rfan)
+            {
+                var arg0value = await rfan.From.Accept(this, context).ConfigureAwait(false);
+                
+                if (arg0value is RecordValue rv)
+                {
+                    ((IMutationCopyField)rv).ShallowCopyFieldInPlace(rfan.Field);

```

Comment:
>((IMutationCopyField)rv) [](http://example.com/codeflow?start=20&length=24)

How do we know this cast will always succeed?

Reply:
The check for arg0.IRContext.IsMutation above this wouldn't be true if this wasn't the case.  But it is fine to add another check to be sure.